### PR TITLE
[FP-07] Financial - Sales Order Performance

### PR DESCRIPTION
**Pull request by arturolinares24**
https://shimoku.atlassian.net/browse/SDT-132

### CODE REVIEW

**Review by rotorrest — CHANGES_REQUESTED** [review]

File: templates/sales_order_performance/utils/utils.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    """
+    Process sales orders performance data from a CSV file.
+
+    Returns:
+    Tuple: A tuple containing Income Total, Spend Total, Net Profit, Average Profit Per Order, and Net Profit by Month.
+    """
+    # # Read sales orders performance data from CSV
+    # df = pd.read_csv("../data/sales_orders_performance.csv")
+
+    # Convert 'order_date' column to datetime format
```
[the remark this answers is not in the document]
Comment (arturolinares24) [reply templates/sales_order_performance/utils/utils.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/99d7eccb1b832f53228cf883da439056b038dbe9  Correction made.

File: templates/sales_order_performance/utils/utils.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+
+import pandas as pd
+import calendar
+import os
+
+def get_data(file_names):
```
[the remark this answers is not in the document]
Comment (arturolinares24) [reply templates/sales_order_performance/utils/utils.py:6]: https://github.com/shimoku-tech/shimoku-app-templates/commit/99d7eccb1b832f53228cf883da439056b038dbe9  Correction made.

File: templates/sales_order_performance/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+shimoku-api-python
```
[the remark this answers is not in the document]
Comment (arturolinares24) [reply templates/sales_order_performance/requirements.txt]: https://github.com/shimoku-tech/shimoku-app-templates/commit/99d7eccb1b832f53228cf883da439056b038dbe9  Correction made.

File: templates/sales_order_performance/main.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+def main():
+    """
+    Main function to initialize and plot the dashboard.
+
+    This script initializes a Shimoku client, deletes existing boards and menu paths,
+    and then creates and plots a new dashboard.
+    """
+    # Load environment variables
+    # load_dotenv()
```
[the remark this answers is not in the document]
Comment (arturolinares24) [reply templates/sales_order_performance/main.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/99d7eccb1b832f53228cf883da439056b038dbe9  Correction made.

File: templates/sales_order_performance/board.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+class Board:
+    """
+    A class representing a Dashboard for displaying various data visualizations.
+
+    Attributes:
+        board_name (str): Name of the dashboard.
+        df (DF): An instance of a DF class for handling data frames.
+        shimoku (Client): An instance of a Client class for Shimoku API interactions.
+        results (dict): A dictionary to store processed data for use in plotting.
+    """
```
[the remark this answers is not in the document]
Comment (arturolinares24) [reply templates/sales_order_performance/board.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/99d7eccb1b832f53228cf883da439056b038dbe9  Correction made.

File: templates/sales_order_performance/README.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+Follow these steps to set up the project locally.
+
+### Clone the repository:
+
+```
+git clone https://github.com/your_username/your_project.git
+```
+```
+cd your_project
```
[the remark this answers is not in the document]
Comment (arturolinares24) [reply templates/sales_order_performance/README.md]: https://github.com/shimoku-tech/shimoku-app-templates/commit/99d7eccb1b832f53228cf883da439056b038dbe9  Correction made.

File: templates/sales_order_performance/README.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+- pip (Python Package Installer), should come installed with Python.
+
+## Installation
+
+Follow these steps to set up the project locally.
+
+### Clone the repository:
+
+```
+git clone https://github.com/your_username/your_project.git
```
[the remark this answers is not in the document]
Comment (arturolinares24) [reply templates/sales_order_performance/README.md]: https://github.com/shimoku-tech/shimoku-app-templates/commit/99d7eccb1b832f53228cf883da439056b038dbe9  Correction made.

**Pull request merged**